### DISCUSSION TITLE
SQL for SUM

### DISCUSSION THREAD

**Discussion by dgotskin** [pocketbase/pocketbase#3753] [Q&A] [answered]
Hello, guys.

Could you please guide me on obtaining the result of this query using your API, version 19.4?

`SELECT SUM(filesize) AS storage_size FROM files;`

Best regards.

**Comment by ganigeorgiev** [comment]
If you are trying to create a [View collection](https://pocketbase.io/docs/collections/#view-collection), the above should work, but you need also an `id` column.

In your case I guess the above query will result in 1 record and you can try something like:

```sql
SELECT 'filesize_sum' as id, SUM(filesize) AS storage_size FROM files;
```

**Reply by dgotskin** [reply]
GPT think so

"If you want to execute SQL queries via the Pocketbase REST API, note that this feature was introduced in version 0.14 or later. Make sure you are using a version that supports SQL queries. You can check the version of Pocketbase you are using and refer to their documentation or GitHub repository for details on how to execute SQL queries." ;)

**Reply by ganigeorgiev** [reply]
It is incorrect. There is no support for executing arbitrary SQL queries with the web APIs. Custom queries are supported only as part of View collections.

**Comment by dgotskin** [comment]
How I can execute such request via https://mysite.net/api/ ... Pocketbase REST API? I try to find this info in documentation and GitHub disq and wad read that sql request should added at 0.14+  versions... Is this correct?

**Reply by ganigeorgiev** [reply, marked as answer]
I'm not sure I follow. The above query is supposed to be defined as [View collection query](https://pocketbase.io/docs/collections/#view-collection).

View collections can be created from the _Admin UI > Collections > New collection_ and select "Type: View collection":

![view_collection_screenshot](https://i.imgur.com/oY06bQC.png)

Once the collection is created, to access its "records" is the same as for the regular collections, eg. `pb.collection("example").getFullList()`.

**Reply by dgotskin** [reply]
Thanks for you work👍
